cluster/src/main/java/org/apache/iotdb/cluster/client/ClientPoolFactory.java

```diff
@@ -0,0 +1,89 @@
+package org.apache.iotdb.cluster.client;
+
+import org.apache.iotdb.cluster.client.async.AsyncDataClient;
+import org.apache.iotdb.cluster.client.async.AsyncMetaClient;
+import org.apache.iotdb.cluster.client.sync.SyncDataClient;
+import org.apache.iotdb.cluster.client.sync.SyncMetaClient;
+import org.apache.iotdb.cluster.config.ClusterConfig;
+import org.apache.iotdb.cluster.config.ClusterDescriptor;
+import org.apache.iotdb.cluster.rpc.thrift.Node;
+import org.apache.iotdb.cluster.rpc.thrift.RaftService;
+
+import org.apache.commons.pool2.impl.GenericKeyedObjectPool;
+import org.apache.commons.pool2.impl.GenericKeyedObjectPoolConfig;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TCompactProtocol;
+import org.apache.thrift.protocol.TProtocolFactory;
+
+import java.time.Duration;
+
+public class ClientPoolFactory {
+
+  protected long waitClientTimeoutMS;
+  protected int maxConnectionForEachNode;
+  private TProtocolFactory protocolFactory;
+
+  ClientPoolFactory() {
+    ClusterConfig config = ClusterDescriptor.getInstance().getConfig();
+    this.waitClientTimeoutMS = config.getWaitClientTimeoutMS();
+    this.maxConnectionForEachNode = config.getMaxClientPerNodePerMember();
+    protocolFactory =
+        config.isRpcThriftCompressionEnabled()
+            ? new TCompactProtocol.Factory()
+            : new TBinaryProtocol.Factory();
+  }
+
+  public static ClientPoolFactory getInstance() {
+    return ClientPoolProviderHolder.INSTANCE;
+  }
+
+  public GenericKeyedObjectPool<Node, RaftService.Client> createSyncDataPool(
+      ClientCategory category) {
+    GenericKeyedObjectPoolConfig config = new GenericKeyedObjectPoolConfig();
```

Comment:
make this config a `field` of  `ClientPoolFactory.java`?

Reply:
Good idea.